File: backend/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Vehicle
    {
        public Guid Id { get; set; }

        [Required]
        [StringLength(20)]
        public string LicensePlate { get; set; } = string.Empty;

        public VehicleStatus Status { get; set; } = VehicleStatus.Active;

        [Required]
        public VehicleType Type { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }

    public enum VehicleStatus
    {
        Active,
        Maintenance,
        Retired
    }

    public enum VehicleType
    {
        Truck,
        Van,
        Trailer
    }
}

File: frontend/src/types/vehicle.ts
export interface Vehicle {
  id: string;
  licensePlate: string;
  status: 'Active' | 'Maintenance' | 'Retired';
  type: 'Truck' | 'Van' | 'Trailer';
  createdAt: string;
  updatedAt: string;
}

export interface VehicleFormValues {
  licensePlate: string;
  status: 'Active' | 'Maintenance' | 'Retired';
  type: 'Truck' | 'Van' | 'Trailer';
}

export interface VehicleResponse {
  vehicles: Vehicle[];
  total: number;
}
